awd/database.py

```diff
@@ -41,92 +40,45 @@ def add_item(cls, doi: str) -> dict[str, Any]:
         return response
 
-    def has_unprocessed_status(self, doi: str) -> bool:
+    def has_unprocessed_status(self) -> bool:
         """Validate that a DOI has unprocessed status in the DOI table."""
-        return DoiProcessAttempt.get(doi).status == Status.UNPROCESSED.value
+        return self.get(self.doi).status == Status.UNPROCESSED.value
 
-    def check_doi_and_add_to_table(self, doi: str) -> None:
-        """Check if DOI should be added to table.
-
-        If not present, add the DOI to the table.  Check for unprocessed status
-        and raise exception if DOI should not be retried. Increment attempts field.
-
-        Args:
-            doi: The DOI to be checked and possibly added to the DOI table.
-        """
-        try:
-            self.get(doi)
-        except DoesNotExist:
-            self.add_item(doi)
-        if not self.has_unprocessed_status(doi):
-            raise UnprocessedStatusFalseError
-        self.increment_attempts(doi)
-
-    @classmethod
-    def attempts_exceeded(cls, doi: str, retry_threshold: int) -> bool:
+    def attempts_exceeded(self, retry_threshold: int) -> bool:
         """Validate whether a DOI has exceeded the retry threshold.
 
         Args:
-            doi: The DOI to be checked.
             retry_threshold: The number of attempts that should be
             made before setting the item to a failed status.
         """
         attempts_exceeded = False
-        item = cls.get(doi)
-        if item.attempts >= retry_threshold:
+        if self.attempts >= retry_threshold:
             attempts_exceeded = True
         return attempts_exceeded
 
-    @classmethod
-    def increment_attempts(cls, doi: str) -> None:
-        """Increment attempts for  DOI item in DOI table.
-
-        Args:
-            doi: The DOI item to be updated in the DOI table.
-        """
-        item = cls.get(doi)
-        logger.debug("Response retrieved from DynamoDB table: %s", item)
-        updated_attempts = item.attempts + 1
-        response = item.update(
-            actions=[
-                cls.attempts.set(updated_attempts),
-                cls.last_modified.set(
-                    datetime.datetime.now(tz=datetime.UTC).strftime(DATE_FORMAT)
-                ),
-            ]
-        )
-        logger.debug("%s attempts updated to: %s", doi, updated_attempts)
-        return response
+    def increment_attempts(self) -> None:
+        """Increment attempts for DOI item in DOI table."""
+        self.attempts += 1
+        self.last_modified = datetime.datetime.now(tz=datetime.UTC).strftime(DATE_FORMAT)
+        self.save()
+        logger.debug("%s attempts updated to: %s", self.doi, self.attempts)
 
-    def set_table_name(self, table_name: str) -> None:
+    @classmethod
+    def set_table_name(cls, table_name: str) -> None:
         """Set table_name attribute.
 
         Args:
             table_name: The name of the DynamoDB table.
         """
```

Comment:
I think this could be a scenario where a comment could be helpful.  Something indicating to future devs that the table name is set dynamically like this for testing / env var restrictions at the time.